Add full form submission e2e test

dpasque opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

## Proposed Changes

Adds a full e2e flow for submitting forms and validating that submission in the new Jetpack admin inbox page.

## Testing Instructions

- [x] Jetpack Simple Desktop
- [x] Jetpack Simple Mobile
- [x] Jetpack Atomic

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Commit:
Add form submission e2e spec

Commit:
Finish validation flow

worldomonation commented on test/e2e/specs/published-content/forms_submission.ts:
```diff
@@ -0,0 +1,140 @@
+const formData = {
+	name: `${ DataHelper.getRandomPhrase() }`,
+	// Making the email unique to each run allows us to filter down to one in the inbox later.
+	email: `test${ DataHelper.getTimestamp() + DataHelper.getRandomInteger( 0, 100 ) }@example.com`,
```
**question:non-blocking** - do you want to switch to using the `EmailClient.getTestEmailAddress` method I implemented in https://github.com/Automattic/wp-calypso/pull/81437?

dpasque replied:
Eh, we can, but just in case it generates any noise, I'd rather not have that fill up mailosaur 😄

worldomonation commented on test/e2e/specs/published-content/forms_submission.ts:
```diff
@@ -0,0 +1,140 @@
+const postTitle = DataHelper.getRandomPhrase();
+
+declare const browser: Browser;
+
```
**nit:blocking** - I think we should write descriptions keywords for each test, to explain briefly what they do.

worldomonation commented on test/e2e/specs/published-content/forms_submission.ts:
```diff
@@ -0,0 +1,140 @@
+		await testAccount.authenticate( page );
+	} );
+
+	describe( 'Create a new post with a Registration Form', function () {
+		it( 'Go to the new post page', async function () {
+			await editorPage.visit( 'post' );
+		} );
+
+		it( 'Enter post title', async function () {
+			await editorPage.enterTitle( postTitle );
+		} );
+
+		it( 'Add a Registration Form block', async function () {
+			await editorPage.addBlockFromSidebar( 'Registration Form', '[aria-label="Block: Form"]', {
+				noSearch: true,
+			} );
+		} );
+
+		it( 'Publish and visit post', async function () {
+			await editorPage.publish( { visit: true } );
+		} );
+	} );
```
**suggestion:non-blocking**- to reduce flakiness, I'm going to propose using an API call with the code representation of the Form block to create a new post, like did in https://github.com/Automattic/wp-calypso/pull/81437/files#diff-49f8202a8c63844a2fc7f8f1f43459d04c59ce3d66abf7516eff3ddc32eb441bR59-R63. 

What do you think?

worldomonation commented on packages/calypso-e2e/src/lib/pages/jetpack-forms-inbox-page.ts:
```diff
@@ -0,0 +1,84 @@
+	 *
+	 * @param {string} text The text to match in the row. Using the name field is a good choice.
+	 */
+	async clickResponseRowByText( text: string ): Promise< void > {
```
**nit:non-blocking** - most methods that only accept one form of parameter to locate the target like this one, we've simply named "clickResponseRow" or similar. The "...ByText" part feels unnecessary to me.

worldomonation commented on packages/calypso-e2e/src/lib/pages/jetpack-forms-inbox-page.ts:
```diff
@@ -0,0 +1,84 @@
+/**
+ * Page representing the Jetpack Forms Inbox page. (The "Inbox", non-"Classic" view for forms responses.)
+ */
+export class JetpackFormsInboxPage {
```
**suggestion:non-blocking** - this seems to be the Feedback > Form Responses page in the sidebar. I thought this page represented the page under SidebarComponent > Inbox which was not the case.

What if this POM was renamed to be simply "FeedbackPage" or "FormResponsesPage"?

Edit: I ran the test and the page this POM represents is this:

![image](https://github.com/Automattic/wp-calypso/assets/6549265/57f18223-9bbb-4b96-98ce-a000ecc575c6)

Which is accessible from SidebarComponent > Feedback. 
However, I see in the view switcher there are two versions - Classic and Inbox. It makes sense now why the current POM has the term "Inbox" in the name.

I still think it's not strictly necessary to call out the "Inbox" part in the naming, and conversely having the focus of the name be "Forms" is confusing as to which feature it maps to in the sidebar.

Given the above, I'd suggest renaming this to `FeedbackInboxPage` as it clearly establishes the feature scope, variant, and the type of POM it is (Page).

worldomonation commented on test/e2e/specs/published-content/forms_submission.ts:
**issue:blocking** - separate feature with flow being tested with double underscores.

worldomonation committed:
test/e2e/specs/published-content/forms__submissions.ts
- rename file

worldomonation committed:
test/e2e/specs/published-content/forms__submissions.ts
- replace EditorPage interactions with an API call.

worldomonation committed:
Rename the POM

worldomonation committed:
test/e2e/specs/published-content/forms__submissions.ts
- rename suite

worldomonation committed:
packages/calypso-e2e/src/jest-playwright-config/environment.ts
- add route aborts.

test/e2e/specs/published-content/forms__submissions.ts
- authenticate after the form submission.

worldomonation committed:
packages/calypso-e2e/src/lib/pages/feedback-inbox-page.ts
- update JSDoc

worldomonation committed:
test/e2e/specs/published-content/forms__submissions.ts
packages/calypso-e2e/src/lib/pages/feedback-inbox-page.ts
- fix typo in method name
